js/src/api/contract/contract.js

```diff
@@ -16,7 +16,9 @@
 
 import Abi from '../../abi';
 import Api from '../api';
-import { isInstanceOf } from '../util/types';
+import { isInstanceOf, isObject } from '../util/types';
+
+const DEFAULT_DEPLOY_GAS = 900000;
```

Comment:
Deployment gas should be a function of length of data, might be best to use estimate gas here.